catalog/ui/catalog-ui-search/src/main/webapp/component/sort-item/sort-item.view.js

```diff
@@ -36,45 +38,141 @@ define([
-                list: [{
-                    label: 'Ascending',
-                    value: 'ascending'
-                }, {
-                    label: 'Descending',
-                    value: 'descending'
-                }],
-                defaultSelection: [this.model.get('direction')]
+            this.handleAttribute();
+            this.turnOnEditing();
+            this.turnOnLimitedWidth();
+
+            this.sortAttribute.currentView.$el.on('change', function (event) {
+                this.handleAttribute();
+            }.bind(this));
+        },
+        turnOffEditing: function () {
+            this.sortAttribute.currentView.turnOffEditing();
+            this.sortDirection.currentView.turnOffEditing();
+        },
+        turnOnEditing: function () {
+            this.sortAttribute.currentView.turnOnEditing();
+            if (!this.$el.hasClass('is-non-directional-sort')) {
+                this.sortDirection.currentView.turnOnEditing();
+            }
+        },
+        turnOnLimitedWidth: function () {
+            this.sortAttribute.currentView.turnOnLimitedWidth();
+            this.sortDirection.currentView.turnOnLimitedWidth();
+        },
+        handleAttribute: function () {
+            var attribute = this.sortAttribute.currentView.getCurrentValue()[0]; //this.model.get('attribute');
+            var ascendingLabel, descendingLabel;
+            if (metacardDefinitions.metacardTypes[attribute] === undefined) {
+                ascendingLabel = descendingLabel = '';
+                this.$el.toggleClass('is-non-directional-sort', true);
+            } else {
+                switch (metacardDefinitions.metacardTypes[attribute].type) {
+                    case 'DATE':
+                        ascendingLabel = 'Earliest';
+                        descendingLabel = 'Latest';
+                        break;
+                    case 'LOCATION':
+                        ascendingLabel = 'Closest';
+                        descendingLabel = 'Furthest';
+                        break;
+                    case 'BOOLEAN':
+                        ascendingLabel = 'True First'; //Truthiest
+                        descendingLabel = 'False First'; //Falsiest
+                        break;
+                    case 'LONG':
+                    case 'DOUBLE':
+                    case 'FLOAT':
+                    case 'INTEGER':
+                    case 'SHORT':
+                        ascendingLabel = 'Smallest';
+                        descendingLabel = 'Largest';
+                        break;
+                    case 'STRING':
+                        ascendingLabel = 'A->Z';
+                        descendingLabel = 'Z->A';
+                        break;
+                    case 'GEOMETRY':
+                    case 'XML':
+                    case 'BINARY':
+                    default:
+                        ascendingLabel = 'Ascending';
```

Comment:
101 is the special case that we don't have a metacard type for the currently selected item (which means we don't know what to put as ascending/descending). So in that case the sort direction (and its label) is completely hidden. (EG, Best Text Match aka relevance)